CHANGES.md

```diff
@@ -18,6 +18,8 @@
     `use_arrow=True` (#265)
 -   Fix errors reading OSM data due to invalid feature count and incorrect
     reading of OSM layers beyond the first layer (#271)
+-   Raise an exception if a data source could not be written when error was not
+    raised by GDAL until closing the data source
```

Comment:
Probably because I'm not a native english speaker, but it took me a few reads before I understood the sentence. Maybe this way it is easier to read?
```suggestion
-   When a datasource is written, also raise an exception if GDAL only gives an error when closing it (#284)
```